saliency/xrai.py

```diff
@@ -0,0 +1,552 @@
+        extra_parameters: an XRAIParameters object that specifies
+                          additional parameters for the XRAI saliency
+                          method. Defaults to None.
+
+    Raises:
+        ValueError: If algorithm type is unknown (not full or fast)
+
+    Returns:
+        np.ndarray: A numpy array that contains the saliency heatmap.
+
+
+    TODO(tolgab) Add output_selector functionality from XRAI API doc
+    """
+    results = self.GetMaskWithDetails(x_value,
+                                      feed_dict=feed_dict,
+                                      baselines=baselines,
+                                      segments=segments,
+                                      extra_parameters=extra_parameters)
+    return results.attribution_mask
+
+  def GetMaskWithDetails(self,
+                         x_value,
+                         feed_dict={},
+                         baselines=None,
+                         segments=None,
+                         extra_parameters=None):
+    """Applies XRAI method on an input image and returns the result saliency
+    heatmap along with other detailed information.
+
+
+    Args:
+        x_value: input value, not batched.
+        feed_dict: feed dictionary to pass to the TF session.run() call.
+                   Defaults to {}.
+        baselines: a list of baselines to use for calculating
+                   Integrated Gradients attribution. Every baseline in
+                   the list should have the same dimensions as the
+                   input. If the value is not set then the algorithm
+                   will make the best effort to select default
+                   baselines. Defaults to None.
+        segments: the list of precalculated image segments that should
+                  be passed to XRAI. Each element of the list is an
+                  [N,M] boolean array, where NxM are the image
+                  dimensions. Each elemeent on the list contains exactly the
+                  mask that corresponds to one segment. If the value is None,
+                  a defaut segmentation algorithm will be applied. Defaults to
+                  None.
+        extra_parameters: an XRAIParameters object that specifies
+                          additional parameters for the XRAI saliency
+                          method. Defaults to None.
+
+    Raises:
+        ValueError: If algorithm type is unknown (not full or fast)
+
+    Returns:
+        XRAIOutput: an object that contains the output of the XRAI algorithm.
+
+    TODO(tolgab) Add output_selector functionality from XRAI API doc
+    """
+    logger.info("Computing IG...")
+    x_baselines = self._make_baselines(x_value, baselines)
+
+    attrs = self._get_integrated_gradients(x_value,
+                                           feed_dict=feed_dict,
+                                           baselines=x_baselines,
+                                           steps=extra_parameters.steps)
```

Comment:
'extra_parameters' is an optional parameter that defaults to None. Calling 'extra_parameters.steps' results in a crash even if the base GetMask(...) method is used.

Reply:
done.